### PR TITLE
Update docs of connect-query for v0.4+

### PR DESCRIPTION
**Pull request by paul-sachs**
Due to recent changes in connect-query (v0.4+), the docs have changed a little. Primarily reducing the core reliance on React specific APIs. Updating the docs to reflect that.

### CODE REVIEW

**Comment by akshayjshah** [comment]
@paul-sachs This is a public repo now - would you mind putting in a short PR description for posterity? If there's nothing notable about the content of the change, it'd be nice to explain why we should be doing this now - was there a recent release or notable change in `connect-query`?

**Comment by akshayjshah** [comment]
@smaye81 is probably the best person to review the substance of this change.

File: docs/web/query/api.mdx [hunk 1]
```diff
@@ -17,32 +17,41 @@ const createQueryService: <Service extends ServiceType>({
...
       },
     },
-    typeName: "connectrpc.eliza.v1.ElizaService",
+    typeName: "your.company.com.example.v1.ExampleService",
   },
 });
 
-const { data, isLoading, ...etc } = useQuery(say.useQuery());
+const example = {
+  ...queryService.say,
```
Comment (smaye81) [inline docs/web/query/api.mdx:44]: I think you want `queryService.example` here (and below) right?

File: docs/web/query/api.mdx [hunk 2]
```diff
@@ -124,51 +133,89 @@ const createUseQueryOptions: (
...
+}) => {
+ mutationFn: (
+  input: PartialMessage<I>,
+  context?: QueryFunctionContext<ConnectQueryKey<I>>
+ ) => Promise<O>;
+ onError?: (error: ConnectError) => void;
+};
+```
+
+`createUseMutationOptions` is intended to be used with TanStack's [`useMutation`](https://tanstack.com/query/v4/docs/react/reference/useMutation) hook. The difference is that `createUseMutationOptions` is not a hook and doesn't read from `TransportProvider` for it's transport.
```
Comment (smaye81) [inline docs/web/query/api.mdx:156]: nit: `...for its transport`

File: docs/web/query/api.mdx [hunk 3]
```diff
@@ -124,51 +133,89 @@ const createUseQueryOptions: (
...
+ queryKey: ConnectQueryKey<I>;
+ queryFn: (
+  context: QueryFunctionContext<ConnectQueryKey<I>, PlainMessage<I>[ParamKey]>
+ ) => Promise<O>;
+ getNextPageParam: GetNextPageParamFunction<O>;
+ onError?: (error: ConnectError) => void;
+};
+```
+
+`createUseInfiniteQueryOptions` is intended to be used with TanStack's [`useInfiniteQuery`](https://tanstack.com/query/v4/docs/react/reference/useInfiniteQuery) hook. The difference is that `createUseInfiniteQueryOptions` is not a hook and doesn't read from `TransportProvider` for it's transport.
```
Comment (smaye81) [inline docs/web/query/api.mdx:181]: same here (`its`)

File: docs/web/query/getting-started.mdx [hunk 4]
```diff
@@ -21,46 +21,55 @@ export const Example: FC = () => {
...
   },
-}).example;
+  typeName,
+} as const;
+
+const $queryService = createQueryService({ service: ExampleService });
+
+export const say = {
+  ...$queryService.say,
+  ...createUnaryHooks($queryService.say),
```
Comment (smaye81) [inline docs/web/query/getting-started.mdx:71]: same as above, should this be `queryService.example`?

Comment (paul-sachs) [reply docs/web/query/getting-started.mdx:71]: Ah yes, nice catch!

Comment (smaye81) [inline docs/web/query/getting-started.mdx:71]: Not sure if you missed changing this one. Should this be `example`?

Comment (paul-sachs) [reply docs/web/query/getting-started.mdx:71]: Ah yeah i def missed that one. Thought I got them all.

**Review by smaye81 — APPROVED** [review]

**Pull request merged**